src/init.cpp

```diff
@@ -209,11 +209,7 @@ void Shutdown()
     // using the other before destroying them.
     if (peerLogic) UnregisterValidationInterface(peerLogic.get());
     if (g_connman) g_connman->Stop();
-    peerLogic.reset();
-    g_connman.reset();
-    if (g_txindex) {
-        g_txindex.reset();
-    }
+    if (g_txindex) g_txindex->Stop();
```

Comment:
Since this is newly added, was it missing completely (independent of the position of pointer resets)?

Reply:
The base destructor was calling it, but yes, imo it was missing.

Reply:
Thanks for fixing then especially!

Reply:
To clarify: right now it would be called twice. Once here and then another time (as belt-and-supenders) in the destructor when the pointer is `reset`. (Just like for conman)

Though, the belt-and-suspenders are not enough and can lead to segfaults occasionally. (See above valgrind output)